cloud/pkg/controller/manager/location.go

```diff
@@ -167,14 +173,65 @@ func (lc *LocationCache) GetAllServices() []v1.Service {
 	return services
 }
 
+// GetService from cache
+func (lc *LocationCache) GetService(name string) (v1.Service, bool) {
+	value, ok := lc.services.Load(name)
+	if !ok {
+		return v1.Service{}, ok
+	}
+	svc, ok := value.(v1.Service)
+	return svc, ok
+}
+
+// AddOrUpdateServicePods in cache
+func (lc *LocationCache) AddOrUpdateServicePods(name string, value []v1.Pod) {
+	lc.servicePods.Store(name, value)
+}
+
+// DeleteServicePods from cache
+func (lc *LocationCache) DeleteServicePods(endpoints v1.Endpoints) {
+	lc.servicePods.Delete(fmt.Sprintf("%s/%s", endpoints.Namespace, endpoints.Name))
+}
+
+// GetServicePods from cache
+func (lc *LocationCache) GetServicePods(name string) ([]v1.Pod, bool) {
+	value, ok := lc.servicePods.Load(name)
+	if !ok {
+		return []v1.Pod{}, ok
+	}
+	pods, ok := value.([]v1.Pod)
+	return pods, ok
+}
+
+// AddOrUpdateEndpoints in cache
```

Comment:
when endpoint updated, maybe some pod remove some label, these pod should be removed from lc.servicePods

Reply:
Nice point ! Yess this is taken care here https://github.com/kubeedge/kubeedge/pull/702/files#diff-55277caec6421504e2848e75c588c384R380